### ISSUE TITLE
[OCI] - Updating Packer to 1.2.3 or 1.2.4 from 1.2.2 problem

### ISSUE THREAD

**Issue by herasimau** [hashicorp/packer#6390]
- Packer version: 1.2.3 or 1.2.4
- Host platform: Linux x64
- Error: "'key_file' must be specified"
- The _simplest example template and scripts_ needed to reproduce the bug.
https://gist.github.com/herasimau/7fe74343f7f767cfb015d82badc8a83d
You can use [DEFAULT] config from here to pass config file:
https://docs.us-phoenix-1.oraclecloud.com/Content/API/Concepts/sdkconfig.htm

Error doesn't present in previous versions such as 1.2.2, 1.2.2

**Comment by prydie** [comment]
It sounds like the validation logic was altered without proper consideration for winrm.

@harveylowndes PTAL.

**Comment by tjones2301** [comment]
I've noticed that this problem occurs when the RSA key is encrypted. Plaintext keys appear to work.

**Comment by SwampDragons** [comment]
I'd like to get a fix into the next patch release; @prydie or @harveylowndes are either of you going to take a look at this or should I?

**Comment by harveylowndes** [comment]
@prydie @SwampDragons it seems that `pass_phrase` is actually mandatory in the builder config if you use a passphrase encrypted key. This is because [ConfigurationProviderFromFileWithProfile](https://github.com/oracle/oci-go-sdk/blob/a2ded717dc4bb4916c0416ec79f81718b576dbc4/common/configuration.go#L220) takes in `privateKeyPassword` as an argument rather than parsing the `configFilePath` to locate it and therefore it is required. This can be easily solved, however I believe it could be useful for the SDK to overload this method where only the `configFilePath` and  `profile` exist as arguments and the `configFilePath` is parsed and also locates the passphrase.

**Comment by SwampDragons** [comment]
So you think just updating the sdk will fix this, then?

**Comment by prydie** [comment]
@harveylowndes Good catch on the SDK issue but are we still blocked on updating due to build issues?

**Comment by harveylowndes** [comment]
@SwampDragons I have tried and tested and it seems to have fixed the issue.
@prydie Unfortunately yes we are. I have opened a PR and it seems the AppVeyor build is still failing.
